cmd/kube-scheduler/app/options/options_test.go

```diff
@@ -331,6 +332,69 @@ profiles:
 		t.Fatal(err)
 	}
 
+	// config with global percentageOfNodesToScore
+	globalPercentageConfig := filepath.Join(tmpDir, "global-percentage-config.yaml")
+	if err := os.WriteFile(globalPercentageConfig, []byte(fmt.Sprintf(`
+apiVersion: kubescheduler.config.k8s.io/v1
+kind: KubeSchedulerConfiguration
+clientConnection:
+  kubeconfig: '%s'
+percentageOfNodesToScore: 50
+profiles:
+- schedulerName: "foo-profile"
+  plugins:
+    reserve:
+      enabled:
+      - name: foo
+      - name: VolumeBinding
+      disabled:
+      - name: VolumeBinding
+`, configKubeconfig)), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
+	}
+
+	// config with profile percentageOfNodesToScore
+	profilePercentageConfig := filepath.Join(tmpDir, "profile-percentage-config.yaml")
+	if err := os.WriteFile(profilePercentageConfig, []byte(fmt.Sprintf(`
+apiVersion: kubescheduler.config.k8s.io/v1
+kind: KubeSchedulerConfiguration
+clientConnection:
+  kubeconfig: '%s'
+profiles:
+- schedulerName: "foo-profile"
+  percentageOfNodesToScore: 100
+  plugins:
+    reserve:
+      enabled:
+      - name: foo
+      - name: VolumeBinding
+      disabled:
+      - name: VolumeBinding
+`, configKubeconfig)), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
+	}
+
+	// config with global and profile percentageOfNodesToScore
+	percentageConfig := filepath.Join(tmpDir, "percentage-config.yaml")
+	if err := os.WriteFile(percentageConfig, []byte(fmt.Sprintf(`
+apiVersion: kubescheduler.config.k8s.io/v1
+kind: KubeSchedulerConfiguration
+clientConnection:
+  kubeconfig: '%s'
+percentageOfNodesToScore: 50
+profiles:
```

Comment:
I would remove these tests. We are already testing the behavior in `scheme_test.go`.

`options_test.go` is about testing configurations that can be overriden by command line flags or are top-level configurations.

Reply:
Removed the new added tests.